### ISSUE TITLE
[Bug]: Misplacement of pictures 

### ISSUE THREAD

**Issue by xuwei152** [3rd/image.nvim#174]
The position of pictures is not corrected, shown as pic below
![Clip_2024-05-29_16-42-17](https://github.com/3rd/image.nvim/assets/51861161/99394760-aac3-4b88-a0d3-433a31c9387f)

The plugin is install with lazy.nvim with the configuration shown below

```lua
{
    "3rd/image.nvim",
    ft = { "markdown" },
    dependencies = {
      {
        "nvim-treesitter/nvim-treesitter",
        build = ":TSUpdate",
        config = function()
          require("nvim-treesitter.configs").setup({
            ensure_installed = { "markdown" },
            highlight = { enable = true },
          })
        end,
      },
    },
    config = function()
      require("image").setup({
        backend = "kitty",
        integrations = {
          markdown = {
            enabled = true,
            clear_in_insert_mode = true,
            download_remote_images = true,
            only_render_image_at_cursor = true,
            filetypes = { "markdown", "vimwiki" }, -- markdown extensions (ie. quarto) can go here
          },
          neorg = {
            enabled = true,
            clear_in_insert_mode = false,
            download_remote_images = true,
            only_render_image_at_cursor = false,
            filetypes = { "norg" },
          },
        },
        max_width = nil,
        max_height = nil,
        max_width_window_percentage = nil,
        max_height_window_percentage = 50,
        kitty_method = "normal",
      })
    end
  },
```

Environment:
- OS: Ubuntu 22.04.4 LTS on Windows 10 x86_64
- tmux next-3.4 in kitty 0.34.1
- NVIM v0.10.0
- commit 645f997d171ea3d2505986a0519755600a26f02f

**Comment by 3rd** [comment]
Hey, do you use any kind of tab bar / breadcrumbs thing?

**Comment by xuwei152** [comment]
Yes, I have installed lualine.nvim. Other loaded plugins listed as follow:

```lazy

  Total: 105 plugins

  Loaded (80)
    ● alpha-nvim 15.92ms  VimEnter
    ● bigfile.nvim 52.18ms  BufReadPre
    ● bufferline.nvim 25.05ms  User FileOpened
    ● cmp-buffer 21.45ms  nvim-cmp
    ● cmp-bufname 29.38ms  VeryLazy
    ● cmp-cmdline 23.37ms  nvim-cmp
    ● cmp-dictionary 21.95ms  VeryLazy
    ● cmp-fuzzy-path 174.08ms  VeryLazy
    ● cmp-git 23.76ms  InsertEnter
    ● cmp-latex-symbols 36.26ms  markdown
    ● cmp-nvim-lsp 17.28ms  nvim-cmp
    ● cmp-omni 16.59ms  InsertEnter
    ● cmp-path 19.76ms  nvim-cmp
    ● cmp_luasnip 33.53ms  nvim-cmp
    ● coc.nvim 80.99ms  InsertEnter
    ● Comment.nvim 16.46ms  User FileOpened
    ● fittencode.nvim 61.95ms  InsertEnter
    ● friendly-snippets 22.39ms  LuaSnip
    ● fuzzy.nvim 1.73ms  cmp-fuzzy-path
    ● fzf 6.24ms  fzf.vim
    ● fzf.vim 13.08ms  start
    ● gitsigns.nvim 22.86ms  User FileOpened
    ● hop.nvim 25.39ms  HopChar1
    ● http.nvim 11.71ms  VeryLazy
    ● image.nvim 174.54ms  markdown
    ● indent-blankline.nvim 13.3ms  User FileOpened
    ● latex-unicoder.vim 1.48ms  markdown
    ● latex.nvim 0.62ms  markdown
    ● lazy.nvim 34.75ms  init.lua
    ● leap.nvim 24.25ms  VeryLazy
    ● lspkind.nvim 4.28ms  start
    ● ltex-client.nvim 14.39ms  VeryLazy
    ● lualine.nvim 69.13ms  VimEnter
    ● LuaSnip 171.2ms 󰢱 luasnip.loaders.from_lua  LuaSnip
    ● mason-lspconfig.nvim 28.7ms  nvim-lspconfig
    ● mason.nvim 15.64ms  mason-lspconfig.nvim
    ● mkdnflow.nvim 24.25ms  markdown
    ● nlsp-settings.nvim 1.32ms  nvim-lspconfig
    ● none-ls.nvim 1.07ms 󰢱 null-ls.rpc  none-ls.nvim
    ● nvim-autopairs 25.8ms  InsertEnter
    ● nvim-cmp 143.54ms  cmp-fuzzy-path
    ● nvim-lastplace 12.53ms  BufRead
    ● nvim-lspconfig 34.92ms  nvim-texis
    ● nvim-markdown 7.33ms  markdown
    ● nvim-navic 10.42ms  User FileOpened
    ● nvim-neoclip.lua 22.76ms  VeryLazy
    ● nvim-texis 41.58ms  start
    ● nvim-treesitter 47.54ms  bigfile.nvim
    ● nvim-treesitter-context 63.22ms  VeryLazy
    ● nvim-ts-context-commentstring 11.58ms 󰢱 ts_context_commentstring.integrations.comment_nvim  /home/xuwei/Trash/lunarvim/lvim/lua/lvim/core/comment.lua
    ● nvim-ufo 22.12ms  VeryLazy
    ● nvim-web-devicons 1.6ms 󰢱 nvim-web-devicons  /home/xuwei/Trash/lunarvim/lvim/lua/lvim/core/breadcrumbs.lua
    ● nvim-window 16.98ms  VeryLazy
    ● nvim_context_vt 3.29ms  VeryLazy
    ● onedarker.nvim 60.71ms  start
    ● plenary.nvim 1.76ms 󰢱 plenary.path  telescope.nvim
    ● promise-async 10.98ms  nvim-ufo
    ● rainbow-delimiters.nvim 24.92ms  start
    ● snippet-converter.nvim 42.66ms  VeryLazy
    ● spelunker.vim 18.91ms  VeryLazy
    ● tabular 38.35ms  InsertEnter
    ● telescope-bibtex.nvim 41.56ms  VeryLazy
    ● telescope-fzf-native.nvim 11.71ms  telescope.nvim
    ● telescope-luasnip.nvim 201.28ms  VeryLazy
    ● telescope.nvim 118.11ms 󰢱 telescope  telescope-luasnip.nvim
    ● todo-comments.nvim 25.2ms  VeryLazy
    ● trouble.nvim 1.7ms 󰢱 trouble  lua
    ● vim-conceal 0.76ms  VeryLazy
    ● vim-cursorword 29.55ms  VeryLazy
    ● vim-custom-surround 2.14ms  VeryLazy
    ● vim-devicons 4.91ms  VeryLazy
    ● vim-diminactive 2.89ms  VeryLazy
    ● vim-easymotion 5.4ms  InsertEnter
    ● vim-illuminate 15.81ms  User FileOpened
    ● vim-surround 2.22ms  InsertEnter
    ● vim-tmux-navigator 1.96ms  VeryLazy
    ● vim-zoom 3.04ms  VeryLazy
    ● vimtex 7.07ms  start
    ● which-key.nvim 55.8ms  VeryLazy
    ● zoomwintab.vim 2.11ms  VeryLazy

```

**Comment by MikeLemo1** [comment]
Maybe an option to give the image an initial global rendering offset would be a good idea it just needs an offset of 2 columns down to be "corrected" assuming because of the barbar + breadcrums extension, also 
The autocmd of the image render at buffer change doesn't work well for me as I have to close and reopen nvim to have the images refreshed in md files, so maybe exposing a function to refresh all images on scene'd useful as well,

BTW It maybe because I don't have sufficient knowledge in lua but how do to access each individual image rendered in markdown? (in what array / through what method etc...

**Comment by 3rd** [comment]
Yes, will add a global offset option, will also try to monitor when something like barbar mounts, the issue if I remember was that it's not there when we check for it.

The integration (https://github.com/3rd/image.nvim/blob/master/lua/image/integrations/markdown.lua) extracts the and creates images, you can use `.get_images()` to query them: https://github.com/3rd/image.nvim/blob/a2a0849e0b3dbed90f9283603cedb683bda5d4d1/lua/image/init.lua#L411

**Comment by xuwei152** [comment]
It is quite weird that some of the pictures are placed properly.
![Clip_2024-06-10_21-53-27](https://github.com/3rd/image.nvim/assets/51861161/0f2f1b69-6ad4-4be9-83b4-b3327e50ac36)

Therefore, the modification of the offset cannot fix the problem.

**Comment by MikeLemo1** [comment]
> It is quite weird that some of the pictures are placed properly. ![Clip_2024-06-10_21-53-27](https://private-user-images.githubusercontent.com/51861161/338215346-0f2f1b69-6ad4-4be9-83b4-b3327e50ac36.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgwMzMyMzcsIm5iZiI6MTcxODAzMjkzNywicGF0aCI6Ii81MTg2MTE2MS8zMzgyMTUzNDYtMGYyZjFiNjktNmFkNC00YmU5LTgzYjQtYjMzMjdlNTBhYzM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEwVDE1MjIxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0N2ZmMjE4ZmQ0MTBjNTk0ZDE5M2UzNDkzZmQwMGI0NDgzNDY4ZWFkZmVhMTIxYmIzZmFkODNjY2UwOGYwNGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.r5vSPkFJL0ApY9Jw_uzDfGwf2LK-EF5n6kPLryidTLY)
> 
> Therefore, the modification of the offset cannot fix the problem.

Are there specific image files you have that display properly?

**Comment by xuwei152** [comment]
Yes, some images are displayed properly. One of them is shown as below.

> It is quite weird that some of the pictures are placed properly. ![Clip_2024-06-10_21-53-27](https://private-user-images.githubusercontent.com/51861161/338215346-0f2f1b69-6ad4-4be9-83b4-b3327e50ac36.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgwMzM2OTIsIm5iZiI6MTcxODAzMzM5MiwicGF0aCI6Ii81MTg2MTE2MS8zMzgyMTUzNDYtMGYyZjFiNjktNmFkNC00YmU5LTgzYjQtYjMzMjdlNTBhYzM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEwVDE1Mjk1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhNjBkYzAyNjM2N2MwZDU5ZGI2Njk4NWZmMDQzYzg3NjdjOWJmMjUyZDNhNmE1NmNkZDA0ZTljYTBjMzZkNDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.O0FvxYF6RqkJeCzvyIKx039elnEZ2JstyD_956JWmw0)
> 
> Therefore, the modification of the offset cannot fix the problem.

I have moved the image that are displayed improperly to other places and then it is placed correctly. I guess the problem is related to text surrounding the image.

**Comment by 3rd** [comment]
> BTW a quick and dirty way to offset the image 2 column down is to go to "~/.local/share/nvim/lazy/image.nvim/lua/image/utils/offsets.lua" and change line 22 from:
> ` local y = 0` 
> to
> ` local y = 2`
> 
> Altough is mostly corrected the problem it requires a new line of spacing in order not to obstruct the content belowe it which is covers half a column below it more than then size of the picture after the 2 column offset but now it's workable 1 extra column that's also serve as spacing is acceptable for now,
> Now displaying svg's I noticed the rendering doesn't take into account the opacity value for some reason...
> any quick fix for that? or that requires diving into the image drivers bit?

ImageMagick is the issue for transparency, I'm working on a complete replacement for it.

**Comment by 3rd** [comment]
> Yes, some images are displayed properly. One of them is shown as below.
> 
> > It is quite weird that some of the pictures are placed properly. ![Clip_2024-06-10_21-53-27](https://private-user-images.githubusercontent.com/51861161/338215346-0f2f1b69-6ad4-4be9-83b4-b3327e50ac36.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTgwMzM2OTIsIm5iZiI6MTcxODAzMzM5MiwicGF0aCI6Ii81MTg2MTE2MS8zMzgyMTUzNDYtMGYyZjFiNjktNmFkNC00YmU5LTgzYjQtYjMzMjdlNTBhYzM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjEwVDE1Mjk1MlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVhNjBkYzAyNjM2N2MwZDU5ZGI2Njk4NWZmMDQzYzg3NjdjOWJmMjUyZDNhNmE1NmNkZDA0ZTljYTBjMzZkNDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.O0FvxYF6RqkJeCzvyIKx039elnEZ2JstyD_956JWmw0)
> > 
> > Therefore, the modification of the offset cannot fix the problem.
> 
> I have moved the image that are displayed improperly to other places and then it is placed correctly. I guess the problem is related to text surrounding the image. 

I think in all cases it's caused by tab bar handling, will look into it.

**Comment by sudo-burger** [comment]
I have a similar issue, apparently caused by concealed wrapped lines. 
The attached markdown file should be a minimal example (you must ensure that the width of the terminal is less than the length of the URLs). 

<img width="468" alt="Screenshot 2024-07-05 at 21 41 41" src="https://github.com/3rd/image.nvim/assets/78477652/0a490439-8efd-4542-b5bc-a14676fbf722">

[images.md](https://github.com/user-attachments/files/16113323/images.md)

**Comment by ficcdaf** [comment]
I can confirm that I am having the same issue with wrapped lines. Disabling wrap fixes the images; but of course, disabling wrap in markdown documents is not preferred.

**Comment by purplebar0** [comment]
I was about to post a new issue, but I'm having the same problem. In the case of `urxvt` and `ueberzugpp`, the images render over text unlike in Kitty (which does it the opposite way). This makes it a lot more apparent:

![Issue preview](https://files.catbox.moe/iryrfg.png)

**Comment by 3rd** [comment]
wrap support has been merged and i have no issues with winbar
closing, please re-open if you find any issues

**Comment by OrionOth** [comment]
Hi, I'm also experiencing this issue! I've tried both Ghostty and Kitty, and I've completely redone my neovim configuration in between my testing attempts (nvchad -> kickstart). The misalignment has been in the same place the entire time.

As you can see, the correct text padding is created to fit the image, however the image is placed one row above and one column to the right of where it should be. I'm not using a tab bar, and when I remove my statusline, it's still in the same place.

```lua
-- using Lazy.nvim
{
  '3rd/image.nvim',
  event = 'VeryLazy',
  dependencies = {
    'nvim-treesitter/nvim-treesitter',
  },
  opts = {
    backend = 'kitty',
    integrations = {
      markdown = {
        enabled = true,
        clear_in_insert_mode = false,
        download_remote_images = true,
        only_render_image_at_cursor = false,
        filetypes = { 'markdown', 'quarto', 'vimwiki' },
      },
    },
    max_width = 100,
    max_height = 12,
    max_height_window_percentage = math.huge,
    max_width_window_percentage = math.huge,
    window_overlap_clear_enabled = true,
    window_overlap_clear_ft_ignore = { 'cmp_menu', 'cmp_docs', '' },
    kitty_method = 'normal',
  },
}
```
- Arch Linux x86_64 kernel `6.15.3`
- Wayland `1.23.1`
- KDE Plasma `6.4`
- kitty `0.42.1`
- ghostty `1.1.3-arch1`
- Neovim `0.11.2`
- Lua `5.4.8`
- LuaJIT `2.1.1748459687`
- image.nvim commit `4c51d62`

![Image](https://github.com/user-attachments/assets/82286d85-b072-4676-817b-be2f0b97323d)

**Comment by goodhumored** [comment]
Btw, folke in his snacks image feature solved this somehow, it works perfectly with his package 

https://github.com/folke/snacks.nvim
